### DISCUSSION TITLE
[BUG] Workflows report failure when action did not run and workflow page looks confusing

### DISCUSSION THREAD

**Discussion by smac89** [community/community#25080] [Actions]
<p>To put this in context, this is the <a href="https://github.com/smac89/gh-actions-samples" rel="nofollow noopener">link</a> to the repository that exhibits this behaviour.</p>
<p>There are three workflows in this repo which are triggered by the <strong>issue</strong> event:</p>
<ul>
<li><a href="https://github.com/smac89/gh-actions-samples/blob/0372d2a5adf45dbe32d8f213b6db4f589198283d/.github/workflows/matrix_with_secrets.yml" rel="nofollow noopener">matrix_with_secrets.yml</a></li>
<li><a href="https://github.com/smac89/gh-actions-samples/blob/0372d2a5adf45dbe32d8f213b6db4f589198283d/.github/workflows/outputs_from_step.yml" rel="nofollow noopener">outputs_from_step.yml</a></li>
<li><a href="https://github.com/smac89/gh-actions-samples/blob/0372d2a5adf45dbe32d8f213b6db4f589198283d/.github/workflows/step_param_from_env.yml" title="step_param_from_env.yml" rel="nofollow noopener">step_param_from_env.yml</a></li>
</ul>
<p>Although they are all triggered by the issue event (opened/reopened), and their jobs are only started conditionally as you can see from the yaml files.</p>
<p>The actions page:</p>
<p><a href="https://github.com/smac89/gh-actions-samples/actions?query=branch%3Amaster+event%3Aissues+actor%3Asmac89" rel="nofollow noopener">https://github.com/smac89/gh-actions-samples/actions?query=branch%3Amaster+event%3Aissues+actor%3Asmac89</a></p>
<p><em>Unfortunately, log searching is not available, so I can’t narrow down the search to a specific date</em></p>
<p>A Gif speaks a million words:</p>
<p><a href="https://imgur.com/a/NkpqNHj" rel="nofollow noopener"><img src="https://i.imgur.com/JscKmZQ.gif" alt="Github workflow bug report December 18, 2019"></a></p>
<p><strong>The issues:</strong></p>
<ul>
<li>
<p>The workflow page looks really congested with actions that did not run at the same time as the last action. As you can see from the gif, only the first three actions ran in that list, but the details page shows about 10 more that had nothing to do with the one I clicked.</p>
</li>
<li>
<p>Another problem is that these workflows run when an issue is created, but after they are stopped because the jobs did not meet the condition to run, they are also reported as having <strong>failed</strong> which is what I don’t understand because the jobs never ran in the first place. Also notice how there is a green checkmark at the top of the page as if the workflow was successful?</p>
</li>
</ul>

**Comment by chrispat** [comment, marked as answer]
<p>We are aware of this issue and are currently working on a fix to accurately represent a skipped job rather than showing it as failed.</p>

**Comment by HDouss** [comment]
<p>I want to add that even an email notification is sent for the failure for every job that did not meet the if condition which is really annoying.</p>
